docs/instrumentation.md

```diff
@@ -0,0 +1,90 @@
+# OpenTelemetry Instrumentation in Bee-Agent-Framework
+
+This document provides an overview of the OpenTelemetry instrumentation setup in the Bee-Agent-Framework.
+The implementation is designed to [create telemetry spans](https://opentelemetry.io/docs/languages/js/instrumentation/#create-spans) for observability when instrumentation is enabled.
+
+## Overview
+
+OpenTelemetry instrumentation allows you to collect telemetry data, such as traces and metrics, to monitor the performance of your services.
+This setup involves creating middleware to handle instrumentation automatically when the `INSTRUMENTATION_ENABLED` flag is active.
+
+## Setting up OpenTelemetry
+
+Follow the official OpenTelemetry [Node.js Getting Started Guide](https://opentelemetry.io/docs/languages/js/getting-started/nodejs/) to initialize and configure OpenTelemetry in your application.
+
+## Instrumentation Configuration
+
+### Environment Variable
+
+Use the environment variable `INSTRUMENTATION_ENABLED` to enable or disable instrumentation.
+
+```bash
+# Enable instrumentation
+export INSTRUMENTATION_ENABLED=true
+# Ignore sensitive keys from collected events data
+export INSTRUMENTATION_IGNORED_KEYS="apiToken,accessToken"
+```
+
+If `INSTRUMENTATION_ENABLED` is false or unset, the framework will run without instrumentation.
+
+### Implementation
+
+The OpenTelemetry middleware is implemented in the `BaseAgent` class in the `run` method
```

Comment:
What about other classes, like `Tool` and `LLM`?

Reply:
I collect events from the middleware function, so I don't depend on other classes.

Reply:
But you can use an `LLM` or a `Tool` without an `Agent`.